First Version of Weather API-v1

carojaspy opened the pull request:
First Version of Weather API, using beego

carlos-rojas-globant committed:
First Version of Weather API, using beego,

carojaspy commented:
@amauryprieto :D

carlos-rojas-globant committed:
Removing Object and User controllers
Removing Object and User models
Removing unused Object and User Routers
Adding Weather Router v1

carlos-rojas-globant committed:
Missing models file added

amauryprieto commented on routers/weatherRouters.go:
```diff
@@ -0,0 +1,7 @@
+package routers
```
@carojaspy what's the purpose of this file?

carojaspy replied:
It was a intent to split the routers as the beego API starter project, but you help me the last week, it was necessary to create the router with **bee generate**, I'll deleted this file from this branch.

amaury-prieto-globant replied:
Great. Please mark this comment as solved.

amauryprieto commented on tests/default_test.go:
```diff
@@ -0,0 +1,61 @@
+	_, file, _, _ := runtime.Caller(1)
+	log.Print(file)
+	// apppath, _ := filepath.Abs(filepath.Dir(filepath.Join(file, ".."+string(filepath.Separator))))
+	apppath := "/mnt/c/Users/carlosalberto.rojas/Desktop/go/src/WeatherAPI/"
```
@carojaspy why aren't you using relative paths?

carojaspy replied:
Because beegon couldn't find the path to my tests, so to don't be stopped i used the absolute path

amaury-prieto-globant replied:
@carojaspy, please try to use a relative path. If I tried to execute the code it will fail.

carojaspy replied:
Solved on this commit: 05579e04c696949f9f6726a45fbb6f9b113e1e0c

amauryprieto commented on tests/default_test.go:
```diff
@@ -0,0 +1,61 @@
+	"testing"
+
+	"github.com/astaxie/beego"
+	. "github.com/smartystreets/goconvey/convey"
```
@carojaspy is there any alternative instead of convey?

carojaspy replied:
I has been reading about testify, I saw that is common to use in go proyects. also looks pretty similar as how other languages implements their tests

amauryprieto commented on time.info:
```diff
@@ -0,0 +1,11 @@
+
```
@carojaspy congrats, clear con concise :)

amauryprieto commented on controllers/weather.go:
```diff
@@ -0,0 +1,70 @@
+
+	// Building response
+	wjson := models.WheatherJSON{}
+	errUn := json.Unmarshal(body, &wjson)
```
@carojaspy do you need an extra variable to handle error?

carojaspy replied:
No, isn't necessary, I should use the error variable declared previously. thanks for the observation.

amauryprieto commented on controllers/weather.go:
```diff
@@ -0,0 +1,70 @@
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Print("Error reading result from api.openweathermap.org")
+		controller.Abort("404")
```
@carojaspy using abort here is correct, however, I dislike that the execution control is in another method. Is there another alternative?

carojaspy replied:
For the v2 I explored to use CustomAbort, but it seems that is almost the same, I'll look for another way.

amauryprieto reviewed (approved):
Congrats. Great job. 

Just small comments. In general, it looks ok to me.

carlos-rojas-globant committed:
Updated README.md: how to run and ow to download the proyect
Fixed Imports to /github.com/carojaspy/ prefix
Updated relative path on default_test.go
Removed weatherRouters.go
